test_to_do_app/test_integration.py

```diff
@@ -1,39 +1,25 @@
+import pymongo
 import pytest
 from dotenv import load_dotenv,find_dotenv
 from todo_app import app
-import requests
+import mongomock
+
 
 @pytest.fixture
-def client(monkeypatch):
+def client():
     file_path = find_dotenv('.env.test')
     load_dotenv(file_path, override=True)
-    monkeypatch.setattr(requests,'get',get_lists_stub)
-    test_app = app.create_app()
+    
+    with mongomock.patch(servers=(('fakemongo.com', 27017),)):
+        test_app = app.create_app()
+        with test_app.test_client() as client:
+            yield client
 
-    with test_app.test_client() as client:
-        yield client
 
-def test_index_page(client):  
+def test_index_page(client): 
     response = client.get('/')
     assert response.status_code == 200
-    assert 'Card 1' in response.data.decode()
```

Comment:
I appreciate that removing this makes the test work, but it also makes it achieve slightly less - i.e. we've lost the confidence that it is actually displaying to-dos, and now just know that it loads _something_
Bringing this line back (e.g. by adding it first) would make this more reliable. To do that you'll need something like updating your patch section to add a card first:
```python
    with mongomock.patch(servers=(('fakemongo.com', 27017),)):
        to_do_items = to_do_list(MongoDB())
        to_do_items.add_card("Card 1")
        test_app = app.create_app()
        with test_app.test_client() as client:
            yield client
```